### PR TITLE
gh-123978: Remove broken time.thread_time() on NetBSD

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
* Issue: gh-123978

### CODE REVIEW

File: Modules/timemodule.c [hunk 1]
```diff
@@ -1485,7 +1485,8 @@ _PyTime_GetThreadTimeWithInfo(PyTime_t *tp, _Py_clock_info_t *info)
 
 #elif defined(HAVE_CLOCK_GETTIME) && \
```
Comment (vstinner) [inline Modules/timemodule.c:1497]: Please add a comment to explain why the clock is disabled on NetBSD, with a reference to the issue (gh-123978).

File: Misc/NEWS.d/next/Library/2024-09-16-12-31-48.gh-issue-123978.z3smEu.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Remove broken :func:`time.thread_time` on NetBSD.
```
Comment (furkanonder) [inline Misc/NEWS.d/next/Library/2024-09-16-12-31-48.gh-issue-123978.z3smEu.rst]:
```suggestion
Remove broken :func:`time.thread_time` and :func:`time.thread_time_ns` on NetBSD.
```
Also we are [removing](https://github.com/python/cpython/blob/a15a584bf3f94ea11ab9363548c8872251364000/Modules/timemodule.c#L1895) the `thread_time_ns` function.

**Review by vstinner — APPROVED** [review]
LGTM

**Comment by vstinner** [comment]
I'm not sure about backporting this change: it can be surprising that a function disappears in a 3.x.y bugfix release. I prefer to keep the broken function in stable versions. (And sadly it's too late for 3.13.0 final.)

**Review by furkanonder — APPROVED** [review]

**Comment by furkanonder** [comment]
LGTM

**Comment by serhiy-storchaka** [comment]
This function is optional. If the underlying OS function is broken, we either try to implement Python API using other functions or do not provide it at all. This is not the first time we have deal with broken OS API.

**Comment by vstinner** [comment]
@serhiy-storchaka:
> This function is optional. If the underlying OS function is broken, we either try to implement Python API using other functions or do not provide it at all. This is not the first time we have deal with broken OS API.

If this is a rationale for backport, well, it's up to you to decide to backport this change or not.

**Pull request merged**